htp/htp_response.c

```diff
@@ -558,25 +558,32 @@ htp_status_t htp_connp_RES_BODY_DETERMINE(htp_connp_t *connp) {
         }
     }
 
+    htp_header_t *cl = htp_table_get_c(connp->out_tx->response_headers, "content-length");
+    htp_header_t *te = htp_table_get_c(connp->out_tx->response_headers, "transfer-encoding");
+
     // Check for "101 Switching Protocol" response.
     // If it's seen, it means that traffic after empty line following headers
     // is no longer HTTP. We can treat it similarly to CONNECT.
     // Unlike CONNECT, however, upgrades from HTTP to HTTP seem
     // rather unlikely, so don't try to probe tunnel for nested HTTP,
     // and switch to tunnel mode right away.
     if (connp->out_tx->response_status_number == 101) {
-        connp->out_state = htp_connp_RES_FINALIZE;
+        if (cl == NULL || te != NULL) {
```

Comment:
I'm not sure I understand this line: if either there is no CL, or if there is a TE, then switch to htp_connp_RES_FINALIZE. Can you explain the CL/TE logic?

Reply:
This is evader case 448
The 101 response code seems to become meaningless (for browsers) by the presence of content-length header (or transfer-encoding)
The logic is to follow the browsers logic even if does not fully make sense to us at first sight.
What do you think ?

(PS : I tried suricata-verify without this patch to find the evader test)

Reply:
If I understand your explanation, the presence of either CL or TE should trigger this new handling. So shouldn't it be
```c
if (cl == NULL && te == NULL) {
    connp->out_state = htp_connp_RES_FINALIZE;
}
```

Reply:
Correct, thanks